_posts/2020-06-12-solidity-0.6-inheritance.md

```diff
@@ -0,0 +1,171 @@
+
+```solidity
+    pragma solidity ^0.6.10;
+    contract A {
+        uint public x;
+        function setValue(uint _x) public virtual {
+            x = _x;
+        }
+    }
+
+    contract B {
+        uint public y;
+        function setValue(uint _y) public virtual {
+            y = _y;
+        }
+    }
+
+    contract C is A, B {
+        function setValue(uint _x) public override(A,B) {
+            A.setValue(_x);
+        }
+    }
+```
+
+Note that you can only override functions if they are marked `virtual`. Additionally any overriding function must be marked `override`. If that again should be overridable, it has to also be marked `virtual`.
+
+`interface` functions are implicitly `virtual` so when implementing an interface you have to explicitly override its functions in the implementation. There is an ongoing discussion on this design [here](https://github.com/ethereum/solidity/issues/8281).
+
+Worth noting that the keyword `super` works as it did before - calls the function one level higher up in the flattened inheritance hierarchy. Also unchanged is `super` is not allowed on `external` functions.
+
+## Interfaces can inherit
+This functionality is new to version 0.6 and allows interface inheritance. The resulting interface is a combination of all inherited interfaces' functions which the contract has to implement.
+
+```solidity
+    pragma solidity ^0.6.10;
+    interface X {
+        function setValue(uint _x) external;
+    }
+
+    interface Y is X {
+        function getValue() external returns (uint);
+    }
+
+    contract Z is Y {
+        uint x;
+        function setValue(uint _x) external override { x = _x; }
+        function getValue() external override returns (uint) { return x; }
+    }
+```
+
+Note that if the contract does not implement all functions it has to be marked as `abstract`.
+
+```solidity
+    pragma solidity ^0.6.10;
+    abstract contract Z is Y {
+        uint x;
+        function setValue(uint _x) external override { x = _x; }
+    }
+```
+
+## Abstract contracts
+With version 0.5 a contract that doesn't implement all of its functions is implicitly made abstract by the compiler:
+
+```solidity
+    pragma solidity ^0.5.17;
+    contract X {
+        function setValue(uint _x) public virtual;
+    }
+```
+
+With 0.6 this differentiation has to be explicit with the compiler generating error `contract X should be made abstract` otherwise
+
+```solidity
+    pragma solidity ^0.6.10;
+    abstract contract X {
+        function setValue(uint _x) public virtual;
+    }
+```
+
+## Public variables safer override for external functions
+Although this feature existed before 0.6 it is made safer for public state variables to override external functions only if the parameter and return types of the function match the getter function of the variable. With version 0.5 it is possible these were allowed to differ as shown in the following example:
```

Comment:
These two sentences sound overly complex to me, I would try to simplify them or split them up if possible?

Reply:
Yes, I managed to abbreviate it to 
>Although this feature existed before 0.6 it is now safer with the added checks that the getter function of the variable (generated by the compiler) matches the parameter and return type of the overriden external function. With version 0.5 it is possible these were allowed to differ as shown in the following example:

Does this sound better?

Reply:
yes!